src/framework/components/anim/component.js

```diff
@@ -0,0 +1,151 @@
+Object.assign(pc, function () {
+
+    /**
+     * @component Anim
+     * @class
+     * @name pc.AnimComponent
+     * @augments pc.Component
+     * @classdesc The Anim Component allows an Entity to playback animations on models.
+     * @description Create a new AnimComponent.
+     * @param {pc.AnimComponentSystem} system - The {@link pc.ComponentSystem} that created this Component.
+     * @param {pc.Entity} entity - The Entity that this Component is attached to.
+     * @property {number} speed Speed multiplier for animation play back speed. 1.0 is playback at normal speed, 0.0 pauses the animation.
+     * @property {boolean} activate If true the first animation will begin playing when the scene is loaded.
+     */
+    var AnimComponent = function (system, entity) {
+        pc.Component.call(this, system, entity);
+    };
+    AnimComponent.prototype = Object.create(pc.Component.prototype);
+    AnimComponent.prototype.constructor = AnimComponent;
+
+    Object.assign(AnimComponent.prototype, {
+
+        loadStateGraph: function(asset) {
+            var data = this.data;
+
+            var graph;
+            var modelComponent = this.entity.model;
+            if (modelComponent) {
+                var m = modelComponent.model;
+                if (m) {
+                    graph = m.getGraph();
+                }
+            }
+
+            var animBinder = new pc.AnimComponentBinder(this, graph);
+            var animEvaluator = new pc.AnimEvaluator(animBinder);
+
+            data.animController = new pc.AnimController(
+                animEvaluator,
+                asset.states,
+                asset.transitions,
+                asset.parameters,
+                this.data.activate
+            );
+        },
+
+        linkAnimAssetToState: function(stateName, asset) {
+            if (!this.data.animController)
+                console.error('linkAnimAssetToState: Trying to link an anim asset to non existing state graph. Have you called loadStateMachineAsset?');

```

Comment:
you'll want to put all console messages in
```
#ifdef DEBUG
#endif
```
pairs

Reply:
Cheers, all logging should be off by default then.